libs/@local/hash-graph-sdk/python/scripts/generate_filter.py

```diff
@@ -0,0 +1,446 @@
+                            keywords=[],
+                        ),
+                        attr="set_cls",
+                        ctx=ast.Load(),
+                    ),
+                    args=[ast.Name(id=class_name)],
+                    keywords=[],
+                )
+            ),
+        ],
+        returns=ast.Subscript(
+            value=ast.Name(id="SelectorPath", ctx=ast.Load()),
+            slice=ast.Name(id=next_class_name),
+        ),
+    )
+
+
+def generate_wildcard_method(
+    token: Enum,
+    name: str,
+) -> ast.FunctionDef:
+    """Generate a method that continues the path with a wildcard selector."""
+    function_name = name
+    while is_keyword_or_builtin(function_name):
+        function_name += "_"
+
+    return create_no_args_method(
+        name=function_name,
+        body=[
+            generate_method_docstring(token, name),
+            ast.Return(
+                value=ast.Call(
+                    func=load_attribute(["PropertiesPath", "from_path"]),
+                    args=[
+                        ast.Call(
+                            func=load_attribute(["self", "path", "push"]),
+                            args=[load_attribute([type(token).__name__, name])],
+                            keywords=[],
+                        )
+                    ],
+                    keywords=[],
+                )
+            ),
+        ],
+        returns=ast.Name(id="PropertiesPath", ctx=ast.Load()),
+    )
+
+
+def generate_continue_method(
+    class_name: str,
+    token: Enum,
+    name: str,
+    next_type: OntologyType,
+) -> ast.FunctionDef:
+    """Generate a method that continues the path with the specified type."""
+    next_class_name = get_class_name(next_type)
+    call_from_path_on = next_class_name
+
+    if class_name == next_class_name:
+        next_class_name = "Self"
+        call_from_path_on = "self"
+
+    function_name = name
+    while is_keyword_or_builtin(function_name):
+        function_name += "_"
+
+    return create_no_args_method(
+        name=function_name,
+        body=[
+            generate_method_docstring(token, name),
+            ast.Return(
+                value=ast.Call(
+                    func=load_attribute([call_from_path_on, "from_path"]),
+                    args=[
+                        ast.Call(
+                            func=load_attribute(["self", "path", "push"]),
+                            args=[load_attribute([type(token).__name__, name])],
+                            keywords=[],
+                        )
+                    ],
+                    keywords=[],
+                )
+            ),
+        ],
+        returns=ast.Name(id=next_class_name, ctx=ast.Load()),
+    )
+
+
+def generate_method(
+    class_name: str,
+    tokens: Enum,
+    method_name: str,
+    *,
+    selector: OntologyType | None = None,
+    wildcard: bool = False,
+    continue_: OntologyType | None = None,
+) -> ast.FunctionDef:
```

Comment:
An overload would probably be good to only allow a single parameter

Reply:
reworked in [c6abea7](https://github.com/hashintel/hash/pull/2878/commits/c6abea70ba0c598f42a7e1651d2bc6161202a104)